gemrb/GUIScripts/InventoryCommon.py

```diff
@@ -549,7 +549,7 @@ def UpdateSlot (pc, slot):
 	SlotType = GemRB.GetSlotType (slot+1, pc)
 	ControlID = SlotType["ID"]
 
-	if not ControlID:
+	if ControlID is None or ControlID is -1:
```

Comment:
ID is a number and `None != 0`, so I don't know what you're trying to achieve here. It also appears to be unsigned, so checking for -1 is odd as well.

Reply:
The -1 here came from GetSlotType, I haven't resolved the cause yet so I included a safety check

```
[Python]: control id:  21
[Python]: control id:  22
[Python]: control id:  23
[Python]: control id:  -1
[Python]: No control found for :  {'Slot': 16, 'ResRef': '-', 'Tip': 0, 'Effects': 5, 'Type': 0, 'ID': -1}
```

Reply:
It's from gemrb/unhardcoded/pst/slottype.2da. So checking for -1 is fine, but the first half still needs to go.